File: 2º/PDOO/ProyectoJAVA/src/civitas/C_impuesto.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author nikitastetskiy
 */
public class C_impuesto extends Casilla{
    private float importe;
    
    C_impuesto(float cantidad, String nombre){
        super(nombre);
        importe = cantidad;
    }
    
    @Override
    void recibeJugador(int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto(actual, todos)) {
            informe(actual, todos);
            todos.get(actual).pagaImpuesto(importe);
        }
    }
    
    @Override
    public String toString() {
        String e = "\n-> Casilla ("+ super.getNombre() + ") " + importe + "$";
        return e;
    }
}

File: 2º/PDOO/ProyectoRUBY/lib/c_impuesto.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative "casilla.rb"
require_relative "jugador.rb"

module Civitas
  class C_impuesto<Casilla
    def initialize(nombre, cantidad)
      super(nombre)
      @cantidad=cantidad
    end
    def  to_s
      devolver=super()
      devolver+="\n\tSu impuesto es de #{@cantidad}"
      devolver+="\n}\n"
      return devolver;
    end  
    def recibeJugador(actual, todos)
      if(actual<=todos.size && actual>0) then
        super.informe(actual, todos)
        return todos[actual].pagaImpuesto(@importe)
      end
      return false
    end
  end
end